contributors/design-proposals/propagation.md

```diff
@@ -150,6 +165,42 @@ non-privileged', an environment check and an WARNING log will be emitted about
 whether propagation mode is supported.
 
 
+## Extra Concerns
+
+@lucab and @euank has some extra concerns about pod isolation when propagation
+modes are changed, listed below:
+
+1. how to clean such pod resources (as mounts are now crossing pod boundaries,
+thus they can be kept busy indefinitely by processes outside of the pod)
+
+1. side-effects on restarts (possibly piling up layers of full-propagation mounts)
+
+1. how does this interacts with other mount features (nested volumeMounts may or
+may not propagate back to the host, depending of ordering of mount operations)
+
+1. limitations this imposes on runtimes (RO-remounting may now affects the host,
+is it on purpose or a dangerous side-effect?)
+
+1. interaction with pod semantics (docker doesn't have a pod concept, but pod-
+aware runtimes may perform additional moves/remounts while preparing a pod)
+
+1. shared/slave flags refer to peer groups. If you want your mounts to be tied to the
```

Comment:
I don't think its implemented, but would love a hostpath/emptydir hybrid that was a per tenant named emptydir. something like:
tenantdir:
  name: foo

maps to something like /var/lib/kubelet/tenantdir/<tenantname>/<name>/

The first pod from a tenant, it gets created like an emptydir.
Pods on a host can come and go from that tenant, but only when the last pod by that tenant goes away does the emptydir delete kick in.

This would make it easy to have daemonsets and deployments land on the same nodes by tenant, and not have to manage exclusive/unique hostpaths to pass data between them.